config-provisioning/src/main/java/com/yahoo/config/provision/CloudAccount.java

```diff
@@ -14,11 +15,25 @@ public class CloudAccount extends PatternedStringWrapper<CloudAccount> {
 
     private static final String EMPTY = "";
     private static final String AWS_ACCOUNT_ID = "[0-9]{12}";
+    private static final Pattern AWS_ACCOUNT_ID_PATTERN = Pattern.compile(AWS_ACCOUNT_ID);
     private static final String GCP_PROJECT_ID = "[a-z][a-z0-9-]{4,28}[a-z0-9]";
+    private static final Pattern GCP_PROJECT_ID_PATTERN = Pattern.compile(GCP_PROJECT_ID);
 
     /** Empty value. When this is used, either implicitly or explicitly, the zone will use its default account */
     public static final CloudAccount empty = new CloudAccount("", EMPTY, "cloud account");
 
+    /** Verifies accountId is a valid AWS account ID and return it unaltered, or throw an IllegalArgumentException. */
+    public static String requireAwsAccountId(String accountId) {
+        Validation.requireMatch(accountId, "AWS account ID", AWS_ACCOUNT_ID_PATTERN);
+        return accountId;
+    }
+
+    /** Verifies accountId is a valid GCP project ID and return it unaltered, or throw an IllegalArgumentException. */
+    public static String requireGcpProjectId(String projectId) {
```

Comment:
Maybe this should return a `CloudAccount` instead?

Reply:
Let me remove the return value as it is not used ATM.  The idea was that it would follow Objects.requireNonNull() and friends that returns its argument.